packages/flutter/lib/src/widgets/table.dart

```diff
@@ -396,44 +402,54 @@ class _TableElement extends RenderObjectElement {
     final Iterator<_TableElementRow> oldUnkeyedRows = _children.where((_TableElementRow row) => row.key == null).iterator;
     final List<_TableElementRow> newChildren = <_TableElementRow>[];
     final Set<List<Element>> taken = <List<Element>>{};
+
+    final List<Element> allOldChildren = <Element>[];
+    final List<Widget> allNewChildren = <Widget>[];
+
     for (final TableRow row in newWidget.children) {
       List<Element> oldChildren;
       if (row.key != null && oldKeyedRows.containsKey(row.key)) {
         oldChildren = oldKeyedRows[row.key]!;
         taken.add(oldChildren);
-      } else if (row.key == null && oldUnkeyedRows.moveNext()) {
+      } else if (oldUnkeyedRows.moveNext()) {
         oldChildren = oldUnkeyedRows.current.children;
```

Comment:
It's possible that add a new keyed row here, 
It’s a bit difficult to handle here because we have to ensure that one-time update of children.

```dart
import 'package:flutter/material.dart';

void main() {
  runApp(
    MaterialApp(
      home: Center(
        child: SizedBox(
          width: 100,
//          height: 100,
          child: W(),
        ),
      ),
    ),
  );
}

class W extends StatefulWidget {
  const W({Key key}) : super(key: key);

  @override
  _State createState() => _State();
}

class _State extends State<W> {
  void _onTap() {
    setState(() => _toggle = !_toggle);
  }

  List<TableRow> _getRows() {
    if (_toggle) {
      return [
        TableRow(key: Key('1'), children: [Text('ke1')]),
        TableRow(children: [Text('unkey')]),
        TableRow(key: Key('2'), children: [Text('ke2')]),
      ];
    } else {
      return [
        TableRow(key: Key('2'), children: [Text('ke2')]),
//        TableRow(key: Key('1'), children: [Text('ke1')]),
        TableRow(children: [Text('unkey')]),
        TableRow(key: Key('3'), children: [Text('ke3')]),
      ];
    }
  }

  bool _toggle = true;
  @override
  Widget build(BuildContext context) {
    return Column(
      children: [
        GestureDetector(onTap: _onTap, child: Text('Tap me')),
        Table(
          children: _getRows(),
        )
      ],
    );
  }
}
```
For the code above, I think the old `key1` should be `dropChild`, and then the new `key3` should be `adoptChild`.
But our current processing is to reuse the old RO and I think there will be no problem with this processing, what do you think?